src/ukify/ukify.py

```diff
@@ -918,6 +935,117 @@ def make_uki(opts):
+            continue
+
+        if name == ".cmdline" and opts.cmdline:
+            cmdline = opts.cmdline
+            if opts.append_cmdline:
+                cmdline = section.get_data().rstrip(b"\x00").decode() + " " + cmdline
+            uki_section = Section.create(name, cmdline, measure=True)
+            additional_sections += [uki_section]
+        else:
+            uki_section = Section.create(name, section.get_data(), measure=True)
+
+        if name == ".linux":
+            linux_section = uki_section
+        else:
+            uki.add_section(uki_section)
+
+    if linux_section is None:
+        raise ValueError("No linux section found! Is this a valid UKI?")
+
+    if opts.signtool == 'sbsign':
+        sign_tool = find_sbsign(opts=opts)
+        sign = sbsign_sign
+        verify_tool = SBVERIFY
+    else:
+        sign_tool = find_pesign(opts=opts)
+        sign = pesign_sign
+        verify_tool = PESIGCHECK
+
+    if sign_kernel is None:
+        # figure out if we should sign the kernel
+        sign_kernel = verify(verify_tool, opts, linux=linux_section.content)
+
+    if sign_kernel:
+        linux_signed = tempfile.NamedTemporaryFile(prefix='linux-signed')
+        linux_signed_path = pathlib.Path(linux_signed.name)
+        sign(sign_tool, linux_section.content, linux_signed_path, opts=opts)
+        linux_section = Section.create(linux_section.name, linux_signed_path, measure=True)
+
+    if pcrpkey := fetch_pcrpkey(opts):
+        additional_sections += [Section.create(".pcrpkey", pcrpkey)]
+
+    pcr_section = call_systemd_measure(uki, linux=linux_section.content, opts=opts)
+
+    # Pop off linux section, so we can put the PCR signature before that
+    pe_linux = pe.sections.pop()
+    if pe_linux.Name != b".linux\x00\x00":
+        raise ValueError("Last section is not a linux section. Is this a valid UKI?")
+
+    pe.__data__ = pe.__data__[:pe_linux.PointerToRawData]
+    pe.__structures__.pop()
+    pe.FILE_HEADER.NumberOfSections -= 1
+    pe.OPTIONAL_HEADER.SizeOfInitializedData -= pe_linux.Misc_VirtualSize
+    pe.OPTIONAL_HEADER.SizeOfImage = round_up(
+        pe.sections[-1].VirtualAddress + pe.sections[-1].Misc_VirtualSize,
+        pe.OPTIONAL_HEADER.SectionAlignment,
+    )
+
+    popped = tempfile.NamedTemporaryFile(prefix="popped_uki")
+    pe.write(popped.name)
+
+    # Start with input UKI without linux section
+    new_uki = UKI(executable=popped.name)
+    for replaced_section in additional_sections:
+        new_uki.add_section(replaced_section)
+
+    if pcr_section:
+        new_uki.add_section(pcr_section)
+
+    new_uki.add_section(linux_section)
+
+    unsigned = tempfile.NamedTemporaryFile(prefix='uki')
+    unsigned_output = unsigned.name
+    pe_add_sections(new_uki, unsigned_output)
+    sign(sign_tool, unsigned_output, opts.output, opts=opts)
+
+    # We end up with no executable bits, let's reapply them
+    os.umask(umask := os.umask(0))
+    os.chmod(opts.output, 0o777 & ~umask)
```

Comment:
The umask here gets never reset or does it? Currently this is not a problem, because ukify will exit shorly after, but the function shouldn't have this side effect.

Reply:
The umask is immediately reset, there's two calls to umask. Really weird implementation detail of umask